tests/datasets/test_transfer.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2018-2019 Faculty Science Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import random
+import string
+from uuid import uuid4
+
+import pytest
+
+from faculty.clients.object import CloudStorageProvider, CompletedUploadPart
+from faculty.datasets import transfer
+
+
+PROJECT_ID = uuid4()
+TEST_PATH = "/path/to/file"
+TEST_URL = "https://example.com/presigned/url"
+OTHER_URL = "https://example.com/other-presigned/url"
+TEST_AWS_UPLOAD_ID = 123
```

Comment:
S3 upload IDs are strings.

Can we also:
```suggestion
TEST_S3_UPLOAD_ID = 123
```